doc/source/guidelines/test_practices.rst

```diff
@@ -0,0 +1,425 @@
+  include tests of the gRPC methods exposed by the proto files and wrapped by
+  the Python library. They would not be expected to test the functionality of
+  the server.
+
+For example, if testing the gRPC method ``GetNode``:
+
+.. code::
+
+   message Node
+   {
+     int32      id = 1;
+     double     x = 2;
+     double     y = 3;
+     double     z = 4;
+   }
+
+   message NodeRequest {
+     int32              num = 1;
+   }
+
+   message NodeResponse {
+     Node               node = 1;
+   }
+
+  service SomeService{
+
+     rpc GetNode(NodeRequest)  returns (NodeResponse);
+     // other methods
+   }
+
+Then your unit test would test the wrapped python function (for example,
+``get_node``).  You might implement the ``get_node`` method with:
+
+.. code:: python
+
+   from ansys.product.service.v0 import service_pb2
+
+   def get_node(self, index):
+       """Return the coordinates of a node for a given index.
+
+       Parameters
+       ----------
+       index : int
+           Index of the node.
+
+       Returns
+       -------
+       tuple
+           Coordinates of the node.
+
+       Examples
+       --------
+       >>> from ansys.product.service import SomeService
+       >>> srv = SomeService()
+       >>> srv.create_node(1, 4.5, 9.0, 3.2)
+       >>> node = srv.get_node(1)
+       >>> node
+       (4.5, 9.0, 3.2)
+
+       """
+       resp = service_pb2.GetNode(index=index)
+       return resp.x, resp.y, resp.z
+
+Your test would be implemented within ``tests/test_nodes.py``:
+
+.. code:: python
+
+   def test_get_node(srv):
+       srv.clear()
+
+       node_index = 1
+       node_coord = 0, 10, 20
+       srv.create_node(node_index, node_coord*)
+       assert srv.get_node(node_index) == node_coord
+
+The goal of the unit test should be to test the wrapping of the
+interface rather than the product or service itself. In the case of
+``GetNode``, this method should have already been tested when designing and
+developing the service.
+
+
+Remote Method Invocation Testing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+In the case of a Remote Method Invocation (RMI)-like method, it is only necessary
+to test the method with a basic case and potentially with any edge cases.
+
+RMI Service Definition:
+
+.. code::
+
+   message SendCommand()
+
+
+Python wrapping:
+
+.. code:: python
+
+   def send_command(command):
+       """Run a command on the server.
+
+       Parameters
+       ----------
+       command : str
+           Command to run on the remote server.
+
+Example test:
+
+.. code:: python
+
+   def test_send_command(srv):
+       output = srv.send_command("CREATE,1")
+       assert "Created 1" in output
+
+Note that this test only validates the command ``"CREATE,1"`` has been
+received, executed, and sent back to the client. It does not validate all
+commands, but nor is it necessary to do this unless there are edge cases
+(e.g. characters that cannot be streamed or dealing with long running
+commands).
+
+
+Testing Framework
```

Comment:
Can we kind of mandate using pytest / unittest for all levels of testing including integration testing. Can we officially frown on use of in-house frameworks/platforms here?

Reply:
@seanpearsonuk I will investigate if we can officially get rid of the internal testing tools such as ARM. I will check that it is not mandatory by any ISO9... norm to have an easy access to testing history somewhere or anything else.